react/ProductCustomizer.js

```diff
@@ -1,54 +1,180 @@
+import AddToCart from './components/Buttons/AddToCart'
+import VariationList from './components/Variation/List'
+import ChangeToppings from './components/Buttons/ChangeToppings'
+import ProductCustomizerService from './utils/ProductCustomizerService'
 
 const isMobile = window.__RUNTIME__.hints.mobile
 
 class ProductCustomizer extends Component {
+  static propTypes = {
+    enableChangeToppings: PropTypes.bool,
+  }
+
+  static defaultProps = {
+    enableChangeToppings: false,
+  }
+
+  state = {
+    total: 0,
+    variations: [],
+    isModalOpen: false,
+    selectedVariations: [],
+    selectionsValidated: false,
+  }
+
+  /**
+  * constructor
+  * Initilize the parent class contructor and parse the data from Schema
+  * @return void
+  */
   constructor() {
     super()
-    this.state = { isModalOpen: false }
-    this.handleOpenModal = this.handleOpenModal.bind(this)
-    this.handleCloseModal = this.handleCloseModal.bind(this)
+
+    const service = new ProductCustomizerService(schema)
+
+    this.state.variations = service.serializeData()
   }
 
-  handleOpenModal() {
+  /**
+  * onHandleRemovePreviousSelectedVariation
+  * Remove the current value from variations list and adds the new one.
+  * @param object variationToRemove
+  * @return bool|void
+  */
+  onHandleRemovePreviousSelectedVariation = (variationToRemove) => {
+    const selectedVariations = this.state.selectedVariations
+
+    const updatedSelectedArray = selectedVariations.filter((currentVariation) => {
+      return variationToRemove === undefined || (variationToRemove.Id !== currentVariation.Id)
+    })
+
+    this.setState({ selectedVariations: updatedSelectedArray })
+  }
+
+  /**
+  * onHandleAddSelectedVariations
+  * Add a new selected variation to array then updates total amount.
+  * @param object variation
+  * @return void
+  */
+  onHandleAddAndCalculateSelectedVariations = (variation) => {
+    const selectedVariations = this.state.selectedVariations
+
+    selectedVariations.push(variation)
+    this.setState({ selectedVariations: selectedVariations })
+
+    this.calculateTotal()
+  }
+
+  /**
+  * calculateTotal
+  * Calculate total amount of selected variations
+  * @return void
+  */
+  calculateTotal = () => {
+    let currentTotal = 0
+    const selectedVariations = this.state.selectedVariations
+
+    if (selectedVariations.length === 0) {
+      return 0
+    }
+
+    for (const selectedVariation of selectedVariations) {
+      currentTotal += selectedVariation.price
+    }
+
+    this.setState({ total: currentTotal })
+  }
+
+  /**
+  * handleOpenModal
+  * Show the product modal
+  * @return void
+  */
+  handleOpenModal = () => {
     this.setState({ isModalOpen: true })
   }
 
-  handleCloseModal() {
+  /**
+  * handleCloseModal
+  * Close the product modal
+  * @return void
+  */
+  handleCloseModal = () => {
     this.setState({ isModalOpen: false })
   }
 
+  handleSubmit = e => {
+    console.log(this.state.selectedVariations)
```

Comment:
Remove `console.log()`

Reply:
Fixed in commit 2d2092c522f572a871eea9c57d10a73b4837ee5f